src/backend/dx12/src/device.rs

```diff
@@ -798,17 +840,60 @@ impl d::Device<B> for Device {
         rp
     }
 
-    fn create_pipeline_layout(&self, sets: &[&n::DescriptorSetLayout], push_constant_ranges: &[(pso::ShaderStageFlags, Range<u32>)]) -> n::PipelineLayout {
+    fn create_pipeline_layout(
+        &self,
+        sets: &[&n::DescriptorSetLayout],
+        push_constant_ranges: &[(pso::ShaderStageFlags, Range<u32>)],
+    ) -> n::PipelineLayout {
         // Pipeline layouts are implemented as RootSignature for D3D12.
         //
         // Each descriptor set layout will be one table entry of the root signature.
         // We have the additional restriction that SRV/CBV/UAV and samplers need to be
         // separated, so each set layout will actually occupy up to 2 entries!
+        //
+        // Root signature layout:
+        //     Root Constants: Register: Offest/4, Space: 0
+        //       ...
+        //     DescriptorTable0: Space: 2 (+1) (SrvCbvUav)
+        //     DescriptorTable0: Space: 3 (+1) (Sampler)
+        //     DescriptorTable1: Space: 4 (+1) (SrvCbvUav)
+        //     ...
+
+        let root_constants = root_constants::split(push_constant_ranges)
+            .iter()
+            .map(|constant| {
+                RootConstant {
+                    stages: constant.stages,
+                    range: (constant.range.start/4)..(constant.range.end/4),
```

Comment:
any checks needed to see if remainder by 4 is not zero?

Reply:
Hm, I should just adjust the API to interpret the range as multiples of u32.